File: Teste_Stage/BackEnd/Teste_Stage/Models/Endereco.cs
using System.ComponentModel.DataAnnotations;

namespace Teste_Stage.Models;

public class Endereco
{
    [Key]
    [Required]
    public int Id { get; set; }

    [MaxLength(50, ErrorMessage = "O tamanho da numero não pode passar de 1 caracteres")]
    public string Logradouro { get; set; }

    [MaxLength(10, ErrorMessage = "O tamanho da numero não pode passar de 1 caracteres")]
    public string Numero { get; set; }

    [MaxLength(50, ErrorMessage = "O tamanho da cidade não pode passar de 1 caracteres")]
    public string Cidade { get; set; }

    [MaxLength(50, ErrorMessage = "O tamanho da estado não pode passar de 1 caracteres")]
    public string Estado { get; set; }

    [MaxLength(2, ErrorMessage = "O tamanho da UF não pode passar de 1 caracteres")]
    public string UF { get; set; }

    // Relacionamento com Candidato
    public virtual Candidato Candidato { get; set; }

    public Endereco()
    {
    }

}

File: Teste_Stage/FrontEnd/Teste_Stage/src/app/common/endereco.ts
export class Endereco {
  id: number;
  logradouro: string;
  numero: string;
  cidade: string;
  estado: string;
  uf: string;

  constructor(
    id: number,
    logradouro: string,
    numero: string,
    cidade: string,
    estado: string,
    uf: string
  ) {
    this.id = id;
    this.logradouro = logradouro;
    this.numero = numero;
    this.cidade = cidade;
    this.estado = estado;
    this.uf = uf;
  }
}
